sql task add

NagorniyAleksandr opened the pull request:
check pls

NagorniyAleksandr committed:
sql task add

NagorniyAleksandr committed:
add some other methods for service

presly808 commented on NagorniyAleksandr/SQLTask/src/main/java/university/container/PropertiesHolder.java:
```diff
@@ -0,0 +1,28 @@
+        try {
+            properties.load(PropertiesHolder.class.getResourceAsStream(PROPERTIES_FILE_PATH));
+        } catch (IOException e) {
+            e.printStackTrace();
```
log info

NagorniyAleksandr replied:
fix

presly808 commented on NagorniyAleksandr/SQLTask/src/main/resources/jdbc.properties:
```diff
@@ -0,0 +1,6 @@
+# MySQL Properties
```
copy jdbc.properties to jdbc.properties and put it to test/resources

define h2 in-memory database for testing

NagorniyAleksandr replied:
done

NagorniyAleksandr replied:
Done one more time) Test DataBase = H2.

NagorniyAleksandr committed:
add logger, override equals/hashcode, implement seervice methods

NagorniyAleksandr committed:
1. add Logger (errors & info for service)
2. all tests now running on test data on other MySQL DataBase.
3. add ability to run test on H2 DataBase BUT negative test fail because H2 has other SQLExceptions hierarchy that MySQL

Commit:
reanimate the project on Windows OS

Commit:
Merge branch 'master' of https://github.com/presly808/ACP15 into addNagorniyModule

Commit:
change logic of exceptions; refactored dao, crud, service

Commit:
change tests for AppDBException, all passed, test DB = MySQL

Commit:
temp commit, errors in H2 base

NagorniyAleksandr committed:
merge master

NagorniyAleksandr committed:
Merge remote-tracking branch 'origin/addNagorniyModule' into addNagorniyModule

NagorniyAleksandr committed:
Done! unit tests based on H2 DataBase

presly808 commented on NagorniyAleksandr/SQLTask/README.md:
```diff
@@ -0,0 +1 @@
+# JDBC
```
Try to describe a business logic of your project 
and put instructions how to run 
e.g. step by step overview

NagorniyAleksandr replied:
done

presly808 commented on NagorniyAleksandr/SQLTask/src/main/java/testRun.java:
```diff
@@ -0,0 +1,38 @@
+import java.util.List;
+
+
+public class testRun {
```
Java Naming, A class name starts form first big letter

NagorniyAleksandr replied:
fixed

presly808 commented on NagorniyAleksandr/SQLTask/src/main/java/university/dao/QueryCreatorImpl.java:
```diff
@@ -0,0 +1,401 @@
+
+            return getStudentsAsListFromResultSet(resultSet);
+        } catch (SQLException e) {
+            log.error(e.getMessage());
```
call logger once, just add additional info as argument

NagorniyAleksandr replied:
Disagrees. I catch SQLException and wrap them in my exceptions. And I would like to see in the log 
1. Exception(SQL Exception) in the class XXX with message "YYY", 
2. Throw new Exception(My exception) from class XXX. 
Then from logs I'll be able track "from->where" my exceptions fly.

presly808 replied:
ok) i dont know that is business logic

presly808 commented on NagorniyAleksandr/SQLTask/src/main/java/university/dao/converter/ToObjectConverter.java:
```diff
@@ -0,0 +1,129 @@
+
+    public static Student getOneStudentFromResultSet(ResultSet resultSet) throws SQLException {
+        if (resultSet.next()) {
+            int studentId = resultSet.getInt(1);
```
Please remember, that your logic depends on right order of table's colums

if someone will change the view of result table you will get a runtime error

presly808 commented on NagorniyAleksandr/SQLTask/src/main/java/university/models/Subject.java:
```diff
@@ -0,0 +1,86 @@
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
if you have the id field, take only it while overriding equals and hashCode

NagorniyAleksandr replied:
fixed

presly808 commented on NagorniyAleksandr/SQLTask/src/main/java/university/service/ServiceImpl.java:
```diff
@@ -0,0 +1,267 @@
+
+import java.util.List;
+
+public class ServiceImpl implements Service {
```
Add validation with Regexp
Example:
```addStudent(Student)```

check fields value of the student object
name, age and so on

NagorniyAleksandr replied:
done

presly808 commented on NagorniyAleksandr/SQLTask/src/test/java/university/dao/crud/CRUDQuerySubjectTest.java:
```diff
@@ -0,0 +1,218 @@
+import static org.hamcrest.core.IsNot.not;
+import static org.junit.Assert.*;
+
+public class CRUDQuerySubjectTest {
```
Several tests repeat beforeClass logic
Create the parent test which will init the schema and data
Parent class will be inherited by child tests

NagorniyAleksandr replied:
fixed

Commit:
Fixed part of the Serhii's comments

Commit:
add README.md

Commit:
add README.md

Commit:
add README.md

Commit:
add validator for inputed objects into Service

Commit:
temp commit

NagorniyAleksandr committed:
change SQL query, refactored log.error

NagorniyAleksandr committed:
temp commit, refactored logger, add alias and validator. need to debug queries...

presly808 commented:
good job
